docs/pallas/gpu/reference.md

```diff
@@ -82,7 +82,96 @@ For more information on how warp scheduling and instruction issue works, we reco
+is not exposed in Pallas:MGPU. While only about a 100MB in size, this memory has considerably
+higher bandwidth than GMEM, and so it is still often recommended to take advantage of it while
+writing high-performance kernels.
+
+Next in line is _shared memory_, or `plgpu.SMEM`. This memory is located directly inside each SM
+and so it is partitioned. Unless block clusters are used (see the section of clusters below),
+each block is only allowed to access its own SMEM allocations.
+
+Finally, the lowest level memory space is the _register memory_. This is where every single value
+(i.e. JAX array) in a Pallas kernel will be located. If the compiler runs out of registers to
+store those arrays, it will insert _spills_, meaning that it will periodically store and reload
+values to memory. Those spills often introduce other significant performance degradations and so
+we recommend avoiding them. The warning messages about spills can be clearly seen in the `ptxas`
+messages during kernel compilation. To make them visible, run with `MOSAIC_GPU_DUMP_PTXAS=1`
+in your environment.
+
+The Blackwell GPU generation, has one additional memory space called _tensor memory_ or `plgpu.TMEM`.
+TMEM is very similar to register memory, only it is explicitly allocated and managed by you.
+It is used to store the MMA accumulator, operand metadata (for sparsity or scaling),
+and optionally the left MMA operand. See the Blackwell MMA section for more information about TMEM.
+
+#### Requesting/allocating memory in specific memory spaces
+
+Kernel inputs or outputs are placed in SMEM by default. If you want to access them as GMEM references
+add `memory_space=plgpu.GMEM` to their `BlockSpec`. If you don't need or want to perform windowing
+it is sufficient to specify `BlockSpec(memory_space=plgpu.GMEM)`.
+
+`SMEM` and `TMEM` can be allocated explicitly in the `scratch_shapes` argument of `pl.pallas_call`,
+or using `pl.run_scoped`. To allocate them, simply call them with the requested shape and dtype.
+For example: `plgpu.SMEM((128, 128), jnp.float16)` will allocate a 128x128 array of float16 elements
+in shared memory.
+
+#### Taking advantage of the L2 cache
+
+While the L2 cache cannot be managed manually, its noticeably higher bandwidth compared to global
+memory makes it worthwhile thinking of. The simplest way to take advantage of it, is to reorder
+the parallel grid dimensions so that invocations that are scheduled in similar time periods also
+access the same input data.
+
+While the CUDA programming model does not guarantee anything about the order in which the blocks
+are assigned to SMs, in recent generations the heuristic seems to simply iterate over the
+`(x, y, z)` CUDA grids in column-major order (i.e. `x` is the fastest changing dimension and
+`z` is the slowest). While Pallas:MGPU does not guarantee how a user-specified grid is mapped to
+the CUDA grid (Pallas supports grids of arbitrary rank, not just up to 3D), you can assume that
```

Comment:
nit: you can assume -> it guarantees? (Sounds like we do want to explicitly promise that.)

Reply:
Not really, because it's based on a CUDA implementation detail NVIDIA is free to change at any time